ledger/test-common/src/main/daml/Test.daml

```diff
@@ -469,20 +469,18 @@ template Delegation
         with
           p: Party
           k: Text
-          expected: Optional (ContractId Delegated)
         do
-          (cid, _) <- fetchByKey @Delegated (p, k)
-          assertMsg "fetch matches" (expected == Some cid)
+          _ <- fetchByKey @Delegated (p, k)
+          pure ()
 
     controller delegate can
       nonconsuming LookupByKeyDelegated: ()
         with
           p: Party
           k: Text
-          expected: Optional (ContractId Delegated)
         do
-          actual <- lookupByKey @Delegated (p, k)
-          assertMsg "lookup matches" (expected == actual)
```

Comment:
Can you elaborate on the removal of these checks?

Reply:
This has to do with how a transaction is validated:
First it is fully interpreted, and only afterwards the original and the re-interpreted transaction are compared. In the case of a test with a false negative key lookup (the participant found no contract id for a given key, but the committer found one), this particular transaction was rejected due to this failed assertion, rather than a failed validation.
Ideally the validation would only produce nodes one by one and compare the produced node directly with the node in the original transaction. This would avoid doing additional work in a failure case. Unfortunately this is not how the validation is currently set up.